### PR TITLE
KEYCLOAK-7858 - OIDC servlet filter adapter OSGi support

### PR DESCRIPTION
**Pull request by dteleguin**
Added documentation

### CODE REVIEW

**Comment by dteleguin** [comment]
@ieugen, could you please review?

File: securing_apps/topics/oidc/java/servlet-filter-adapter.adoc
Comment (ieugen) [inline securing_apps/topics/oidc/java/servlet-filter-adapter.adoc]: > First, the adapter needs to be registered as an OSGi component 

First, the adapter needs to be registered as a Servlet filter for OSGi HTTP Service. You can do that in multiple ways. The most common ways are via bundle activator or via declarative services.
We recommend using declarative services since it simplifies the process of dynamically registering and un-registering the filter.

File: securing_apps/topics/oidc/java/servlet-filter-adapter.adoc
Comment (ieugen) [inline securing_apps/topics/oidc/java/servlet-filter-adapter.adoc]: I would add: 

The above snippet uses OSGi declarative service specification to expose the filter as a declarative service component and as an OSGI service under `javax.servlet.Filter` class. 
Once the class is published in the OSGi service registry it is going to be picked up by OSGi HTTP Service implementation and used for filtering requests for the specified servlet context. This will trigger Keycloak adapter for every request that matches the servlet context path + the filter path. 

Since the component is using ConfigAdmin service you can configure it's properties dynamically like this:

File: securing_apps/topics/oidc/java/servlet-filter-adapter.adoc
Comment (ieugen) [inline securing_apps/topics/oidc/java/servlet-filter-adapter.adoc:159]: Another popular HTTP Service implementation is https://ops4j1.jira.com/wiki/spaces/paxweb/overview[OPS4j Pax-WEB] . If you are using Karaf, most likely you are using pax-web.

File: securing_apps/topics/oidc/java/servlet-filter-adapter.adoc
Comment (hmlnarik) [inline securing_apps/topics/oidc/java/servlet-filter-adapter.adoc]: Please specify minimum OSGi release known to work.

**Comment by dteleguin** [comment]
OK guys, I'm giving up. Travis has persistent problems connecting to *.apache.org, thus the link tests would always fail. This is not something I can fix unfortunately.

**Comment by dteleguin** [comment]
Another issue with variable checking - the test would fail for an empy Java construct inside a `[source]` block:
```java
public class Foo {
}
```
As a workaround, I had to add a `//` to the class body. But should the checker indeed stick its nose into `[source]` blocks, unless variable substitution is requested explicitly with `sub=...`?

**Comment by matthewhelmke** [comment]
@dteleguin Thank you for the submission.
@ieugen Thank you for the additional comments and suggestions.
@stianst Can you please have someone from engineering take a look into the build issues here?

I will give this a final review from a technical writing perspective once the dust settles.

**Comment by ieugen** [comment]
@matthewhelmke It would be awesome to have this in Keyclaok 4.4 :D .

File: securing_apps/topics/oidc/java/servlet-filter-adapter.adoc
Comment (hmlnarik) [inline securing_apps/topics/oidc/java/servlet-filter-adapter.adoc:76]: Please add a note for Karaf with Pax-Web users to prefer using Fuse 6/7 adapters (Fuse is based on Karaf) (and refer the relevant sections in this guide).

**Review by hmlnarik — CHANGES_REQUESTED** [review]
@dteleguin Technically looks good, thank you. I need to test the details before giving final approval before language review. Still there are few items to address already, feel free to address them.

**Comment by dteleguin** [comment]
@hmlnarik, @matthewhelmke I've addressed the issues you've pointed out and included @ieugen's suggestions as well.

**Comment by dteleguin** [comment]
Please see the most recent comments under https://github.com/travis-ci/travis-ci/issues/6593.
The known workaround it to use `sudo: required`.

**Comment by hmlnarik** [comment]
@dteleguin Thanks for the changes. Could you please squash the commits? Then it will be ready to merge

**Comment by dteleguin** [comment]
@hmlnarik done

**Review by hmlnarik — APPROVED** [review]

**Comment by hmlnarik** [comment]
@dteleguin Thank you

**Pull request merged**
